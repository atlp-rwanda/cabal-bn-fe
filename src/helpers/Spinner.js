import React from 'react';
import { Box, CircularProgress } from '@mui/material';

export default () => (
  <Box sx={{ position: 'absolute', left: '47.5%', top: '45%' }}>
    <CircularProgress size={60} />
  </Box>
);
