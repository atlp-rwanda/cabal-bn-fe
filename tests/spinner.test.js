import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from '../src/helpers/Spinner';

describe('Navbar', () => {
  it('should render the spinner on click', () => {
    const component = renderer.create(<Spinner />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
      tree.props.close;
    });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
