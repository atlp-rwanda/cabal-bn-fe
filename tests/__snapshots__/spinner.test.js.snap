// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the spinner on click 1`] = `
<div
  className="MuiBox-root css-1kwo37v"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 60,
        "width": 60,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;

exports[`Navbar should render the spinner on click 2`] = `
<div
  className="MuiBox-root css-1kwo37v"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 60,
        "width": 60,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;
